plugins/src/main/kotlin/extension/AnvilExtensions.kt

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2024 New Vector Ltd.
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ * Please see LICENSE in the repository root for full details.
+ */
+
+package extension
+
+import org.gradle.api.Project
+import com.squareup.anvil.plugin.AnvilExtension
+import org.gradle.accessors.dm.LibrariesForLibs
+import org.gradle.kotlin.dsl.the
+
+/**
+ * Setup Anvil plugin with the given configuration.
+ * @param generateDaggerCode whether to enable general Dagger code generation using Kapt
+ * @param generateDaggerFactoriesUsingAnvil whether to generate Dagger factories using Anvil instead of Kapt
+ */
+fun Project.setupAnvil(
+    generateDaggerCode: Boolean = false,
+    generateDaggerFactoriesUsingAnvil: Boolean = true,
+) {
+    val libs = the<LibrariesForLibs>()
+    // Apply plugins and dependencies
+    applyPluginIfNeeded("com.squareup.anvil")
```

Comment:
```suggestion
    applyPluginIfNeeded(libs.plugins.anvil)
```

with change on `applyPluginIfNeeded` to

```kotlin
private fun Project.applyPluginIfNeeded(plugin: Provider<PluginDependency>) {
    val pluginId = plugin.get().pluginId
    if (!pluginManager.hasPlugin(pluginId)) {
        pluginManager.apply(pluginId)
    }
}
```